src/main/java/gomedic/logic/parser/findcommandparser/FindDoctorCommandParser.java

```diff
@@ -0,0 +1,66 @@
+package gomedic.logic.parser.findcommandparser;
+
+import gomedic.commons.core.Messages;
+import gomedic.logic.commands.findcommand.FindDoctorCommand;
+import gomedic.logic.parser.Parser;
+import gomedic.logic.parser.exceptions.ParseException;
+import gomedic.model.util.DepartmentContainsKeywordsPredicate;
+import gomedic.model.util.NameContainsKeywordsPredicate;
+import gomedic.model.util.PhoneNumberContainsKeywordsPredicate;
+import static gomedic.logic.parser.CliSyntax.DEPARTMENT;
+import static gomedic.logic.parser.CliSyntax.NAME;
+import static gomedic.logic.parser.CliSyntax.PHONE_NUMBER;
+
+
+import java.util.Arrays;
+
+/**
+ * Parses input arguments and creates a new FindDoctorCommand object
+ */
+public class FindDoctorCommandParser implements Parser<FindDoctorCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindDoctorCommand parse(String args) throws ParseException {
```

Comment:
Similar to the parsing methods previously, please refer to the other existing classes alerady